File: app/src/main/java/com/example/navigationdrawerapp/BaseViewHolder.java
package com.example.navigationdrawerapp;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

public abstract class BaseViewHolder extends RecyclerView.ViewHolder {

    private static String TAG = BaseViewHolder.class.getSimpleName();
    private final DrawerListener listener;

    public BaseViewHolder (View itemView, DrawerListener listener) {
        super(itemView);
        this.listener = listener;
    }

    protected DrawerListener getListener() {
        return listener;
    }

    abstract void bindDataToViewHolder(DrawerModel model, int position);
}

File: app/src/main/java/com/example/navigationdrawerapp/BaseViewHolder.kt
package com.example.navigationdrawerapp

import android.view.View
import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder : RecyclerView.ViewHolder {

    companion object {
        private val TAG = BaseViewHolder::class.java.getSimpleName()
    }

    private val listener : DrawerListener

    constructor(itemView : View, listener : DrawerListener) : super(itemView) {
        this.listener = listener
    }

    protected fun getListener() : DrawerListener {
        return listener
    }

    abstract fun bindDataToViewHolder(model : DrawerModel, position : Int)
}